vault/resource_userpass_user_test.go

```diff
@@ -0,0 +1,138 @@
+		Providers:    testProviders,
+		PreCheck:     func() { testutil.TestAccPreCheck(t) },
+		CheckDestroy: testAccUserpassUserCheckDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccUserpassUserConfig_basic(backend, username, password, []string{"admin", "security"}),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resName, "id", "auth/"+backend+"/users/"+username),
+					resource.TestCheckResourceAttr(resName, "backend", backend),
+					resource.TestCheckResourceAttr(resName, "username", username),
+					resource.TestCheckResourceAttr(resName, "password", password),
+					resource.TestCheckResourceAttr(resName, "token_policies.#", "2"),
+					resource.TestCheckResourceAttr(resName, "token_policies.0", "admin"),
+					resource.TestCheckResourceAttr(resName, "token_policies.1", "security"),
+				),
+			},
+			{
+				Config: testAccUserpassUserConfig_basic(backend, username, password, []string{"updated_policy"}),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resName, "id", "auth/"+backend+"/users/"+username),
+					resource.TestCheckResourceAttr(resName, "backend", backend),
+					resource.TestCheckResourceAttr(resName, "username", username),
+					resource.TestCheckResourceAttr(resName, "password", password),
+					resource.TestCheckResourceAttr(resName, "token_policies.#", "1"),
+					resource.TestCheckResourceAttr(resName, "token_policies.0", "updated_policy"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccUserpassUser_importBasic(t *testing.T) {
+	backend := acctest.RandomWithPrefix("userpass")
+	resName := "vault_userpass_user.user"
+	user := "u3er_name"
+	password := "pa33_word"
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testutil.TestAccPreCheck(t) },
+		Providers: testProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccUserpassUserConfig_basic(backend, user, password, []string{"security", "admin"}),
+			},
+			{
+				ResourceName:            resName,
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"password"},
+			},
+		},
+	})
+}
+
+func TestAccUserpassHelpers(t *testing.T) {
+	t.Run("Test helper functions", func(t *testing.T) {
+		expectedPath := "auth/userpass/users/blm_hemu"
+		expectedBackend := "userpass"
+		expectedUsername := "blm_hemu"
+
+		actualPath := userPath(expectedBackend, expectedUsername)
+		actualBackend := backendFromPath(expectedPath)
+		actualUsername := usernameFromPath(expectedPath)
+
+		if actualPath != expectedPath {
+			t.Fatalf("expected path '%s', got: '%s'", expectedPath, actualPath)
+		}
+
+		if actualBackend != expectedBackend {
+			t.Fatalf("expected backend '%s', got: '%s'", expectedBackend, actualBackend)
+		}
+
+		if actualUsername != expectedUsername {
+			t.Fatalf("expected username '%s', got: '%s'", expectedUsername, actualUsername)
+		}
+	})
+}
+
+func testAccUserpassUserCheckDestroy(s *terraform.State) error {
```

Comment:
We don't need this since we're checking that the mount was destroyed using an existing helper function